### ISSUE TITLE
Any way to get around this?

### ISSUE THREAD

**Issue by ma7555** [zstachniak/Memory-Leak-in-MySQL-Connector#1]
I am suffering from the same issue, did you find anyway to solve this?

**Comment by zstachniak** [comment]
You _can_ get around it by enclosing every insert inside of a multiprocessing thread. I coded a solution that did work, but it was company-specific so I can't share that code. It's a last-resort type of fix, though, because you lose so much in the process.

- you no longer have one connection, but are spawning multiple database connections (I avoided async multiprocessing in order to not have the additional worry of database connection pooling)
- you lose all built-in ability to checkpoint and revert to a checkpoint. That means that every insert must be _committed_. No more transaction-level processing. 
- consider that an "error" in one of your inserts will now no longer be surrounded within a transaction that can be rolled back. You will now have to implement error handling that can manually roll back (see next point).
- If you want the ability to roll back to a point prior to an insert, you are now also tasked with tracking every insert that you spawn in your multiprocessing threads (depending on the database, that can be either very simple or very hard). You will also have to write delete methods to delete data as needed, also paying attention to foreign keys. _Note how much more dangerous this is than restoring to a checkpoint!_
- The additional commits (both inserts and deletes) will unnecessarily tax your database with crud. If you are inserting large amounts of data (which I'm assuming you are if you're having this issue), you may need to periodically defrag your tables using OPTIMIZE TABLE.

Long story short, I don't recommend it. You'll basically lose everything that makes a database worth using. The admins still have yet to respond to my bug report, so what are the other options?

Don't use the official mysql connector? Don't use Python? Don't use MySQL? I'd suggest any of those before trying to solve with multiprocessing.

**Comment by ma7555** [comment]
I have thought about making every insert in a new thread that would die after completion too. I am using pandas to_sql which suffers from the same bug (as it is just an api for my_sql connector).
I however have updated all my packages (conda update --all) and surprisingly the issue seems to be gone now... I have kept the snapshots going for 5 minutes now and can't seem to have any problems.

I stopped monitoring for now as I had to go and will check back later but i think it is fine now.
